### ISSUE TITLE
Please consider change version tagging convention

### ISSUE THREAD

**Issue by kloczek** [openssl/openssl#11647]
Latest version is `OpenSSL_1_1_1g`. It would be really nice to use just `1_1.1g`.

Using current convention is causing that git autogenberated tar ball from https://github.com/openssl/openssl/archive/OpenSSL_1_1_1g.tar.gz l has `openssl-OpenSSL_1_1_1g/` base directory.

Also using tag `1_1.1g` would allow better integration with all automated packaging systems as all package managers does not allow use underscores in version string.

If change current convention may break something please consider add duplicates of the tags in format which I'm proposing.

**Comment by levitte** [comment]
We have actually considered that or very similar, and codified it for 3.0.  We have talked about it for the older series (including 1.1.1) as well, but haven't acted on it yet.

The way we've codified it for 3.0 is that the release tag will be something like `openssl-3.0.0` when we make the final release...  there already is a tag `openssl-3.0.0-alpha1` for the alpha release we made a few days ago.

I would maintain that `1.1.1g` isn't enough, it should be `openssl-1.1.1g`.  The very simple reason is that if the OpenSSL history is placed in a repo together with other projects, it's still possible to distinguish what tag belongs where.  Other open source projects to exactly this...

**Comment by kloczek** [comment]
Openssl repository holds only openss code so using `openssl-<version>` will be redundant and will cause that auto generated by git tar balls will have base directory `openssl-openssl-<version>` as git automatically adds to tag name repository name.

Please consider adding duplicate tags on the same hashes with only `<version>` if it is any technical reason in exist in context of any automation currently used by openssl maintainers.

Just look around on repositories wit will maintained set of projects gnome projects where is used that kind convention which is the best on interaction wit any automation taking tag literally as package version which needs to be packaged.

Please do not use suffixes like `'alpha`, `beta`. Just use `<last.version>.99` and `<last.version>.99.1` and finally `<new.version>` or similar to allow easy upgrade packaged non-release tagged code to release one. Using suffixes like `alpha` is always tricky on packaging as `<version>alpha` is always greater than `<version>`.

Many projects uses that kind convention and it is the best convention on interaction with automation packaging projects from git tags.

I'm working now on automation which my produce you guys automatic report on testing new version on building all packages which will be using openssl. With that you will have automatic confirmation that your changes are not afflicting any package in typical OS distribution.

Have such confirmation should be for you great value so IMO it would be good to agree on some **well defined interface** which would allow produce such cooperation to bring you such report automatically or on demand. Openssl is quite important project and I don't think that any of you would be able to provide such easy of testing new version as you probably not working on whole OS distribution(s).

Please consider above rations to choose convention which I've described. This details is not only about style or personal preferences.

**Comment by levitte** [comment]
Er, so this is a *Gnome* convention, do I understand that correctly?

**Comment by kaduk** [comment]
> there already is a tag `openssl-3.0.0-alpha1` for the alpha release we made a few days ago

@levitte I don't see one in either the main repo or github.  Perhaps it didn't get pushed?

**Comment by levitte** [comment]
> @levitte I don't see one in either the main repo or github. Perhaps it didn't get pushed?

It's there now.  It should have been pushed when we released, but maybe we missed that step...

**Comment by kloczek** [comment]
> Er, so this is a Gnome convention, do I understand that correctly?

No this in not only gnome. Probably +95% of all git using projects is using  just`<version>`  or `v<version>` (git strips leading 'v`).
Check all perl modules, qt, kde, xfce, mate, xorg, freedesktop.

Simple there is no in reality any logical resons to use anything than just plan version string with dots.

**Comment by richsalz** [comment]
I agree that tagging things with openssl-, in the openssl repository, is silly.
Using alpha, beta is much better than using .99 (which also hides the alpha/beta difference).

**Comment by kloczek** [comment]
I would not say "silly". More likely "not logical" or "redundanmt" :)

I found trat some packages are using speciall versioning convention for non prod/devel/alpha/nbeta. Siple if lets say version consist from sections like `major.minor.update` some packkeges uses `major.minor.update.devel`.

To be honest I like that convention because it allows easy distingus non-prod/non-release version and allows easy upgrade from `major.minor.update` to `major.minor.update.alpha` . and than to major.minor.newupdate`.

**Comment by mattcaswell** [comment]
I could live with just the version number (TBH I'm forever switching from one branch to another, so it's not a bad idea for the simple reason that its less characters to type!!).

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4  dev, barring further input

**Comment by kloczek** [comment]
Currently is used as tag patter `openssl-<version.with.dots>`.
Still it can be used just `<version.with.dots>` or `v<version.with.dots>` (leading `v` is automatically skipped by github on autogenerate tar ball from git tag)

**Comment by nhorman** [comment]
ok, marking as to be considered for a future minor release

**Comment by kloczek** [comment]
Thank you 👍

**Comment by levitte** [comment]
I dunno.  It does get up my nose how much github gets to dictate the format of tags, it's really quite annoying...  and triggers the rebel in me.  So I'm ambivalent.

**Comment by kloczek** [comment]
Every other REST/web interface forces those tings as well.

**Comment by levitte** [comment]
Not quite.  I just read through [Gitlab's "Tutorial: Automate releases and release notes with GitLab"](https://about.gitlab.com/blog/2023/11/01/tutorial-automated-release-and-release-notes-with-gitlab/), and it looks like their *default* release generator works like you way, but that you can make your own custom pipeline for it as well.

But anyway...  if we're to change our tags *again*, our [branch policy](https://github.com/openssl/technical-policies/blob/master/policies/branch-policy.md) must be updated, and a few scripts along with it...

**Comment by t8m** [comment]
Can we add both tags? Or would that just confuse things?

**Comment by t8m** [comment]
So I disagree with this feature request at least if implemented in the simplest way.

**Comment by kloczek** [comment]
> Also IMO unless we decide to drop generating our own tarballs in the release process it is actually good that the directory structure of the autogenerated github tarball is different from the _official tarball_.

- githup provides PGP/GPG signing commits and tags
- on use tar balls autogenerated from git tag it is 100% guarantee that all patches generated from commits added after release tag will be possible to apply without rejections.
